nml/actions/action2var_variables.py

```diff
@@ -562,8 +562,14 @@ def industry_town_count(name, args, pos, info):
     return (args[0], extra_params)
 
 def industry_cargotype(name, args, pos, info):
-    from nml.expression.functioncall import builtin_cargotype
-    return (builtin_cargotype(name, args, pos), [])
+    if len(args) != 1:
+        raise generic.ScriptError(name + "() must have 1 parameter", pos)
+    if not isinstance(args[0], expression.StringLiteral) or args[0].value not in global_constants.cargo_numbers:
+        raise generic.ScriptError(
+            "Parameter for {}() must be a string literal that is also in your cargo table", pos
+        )
+    num = expression.ConstantNumeric(global_constants.cargo_numbers[args[0].value])
+    return (num, [])
```

Comment:
Ok, so this is *increasing* duplication...I'll try to find a better way to fix this.

(also, I just realized I forgot to format the format-string here...)

Reply:
Can't import `function_table` from `functioncall` and use `function_table["cargotype"](name, args, pos)` ?

Reply:
I couldn't then, because `name` wouldn't have been in the lookup table. Sorted now though.